applications/MeshMovingApplication/custom_utilities/mesh_velocity_calculation.h

```diff
@@ -0,0 +1,90 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license:
+// kratos/license.txt
+//
+//  Main authors:    Philipp Bucher
+//
+
+#if !defined(KRATOS_MESH_VELOCITY_COMPUTATION_H_INCLUDED)
+#define KRATOS_MESH_VELOCITY_COMPUTATION_H_INCLUDED
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "includes/define.h"
+#include "includes/model_part.h"
+#include "includes/mesh_moving_variables.h"
+#include "utilities/time_discretization.h"
+
+namespace Kratos {
+namespace MeshVelocityCalculation {
+
+template<class TGenAlpha>
+void CalculateMeshVelocitiesGeneralizedAlpha(ModelPart& rModelPart,
```

Comment:
Can this go in the cpp file?

Reply:
This is a templated function
I dont know (tried several times and failed always) how to instantiate it in the cpp
For classes it works but I didn't get it working with functions
Do you know if this is possible?

Reply:
I found an even better solution, pls have a look :)